### PR TITLE
Meshgrid API Changes

### PR DESCRIPTION
**Pull request by luitjens**
Update Meshgrid APIs to work with ND tensors and return operators for the output tensors.

### CODE REVIEW

File: include/matx/generators/meshgrid.h [hunk 1]
```diff
@@ -31,89 +31,89 @@
...
 
-        Meshgrid_X(std::array<T, 3> x, std::array<T, 3> y) : x_(x), y_(y) {}
+          __MATX_INLINE__ std::string str() { return "meshgrid"; }
 
-        inline __MATX_DEVICE__ T operator()(index_t i, index_t j) const
-        {
-          return x_[0] + j * (x_[1] - x_[0]) / (x_[2] - 1);
-        }
+          __MATX_INLINE__ MeshGridOp(T1 t1, std::array<index_t, RANK> shape) : t1_(t1), shape_(shape) {
+            static_assert(shape.size() == RANK );
```
Comment (cliffburdick) [inline include/matx/generators/meshgrid.h:56]: Maybe add a second argument to each of these saying why the assert failed

File: include/matx/generators/meshgrid.h [hunk 2]
```diff
@@ -31,89 +31,89 @@
...
+      };
+
+    template <typename... Ts, int... I>
+      __MATX_INLINE__  auto meshgrid_impl(std::integer_sequence<int, I...>, Ts&... ts) {
+        auto constexpr RANK = (int)sizeof...(Ts);
+
+        // construct shape from size of each rank1 tensor
+        std::array<index_t, RANK> shape{ts.Size(0)...};
+
+        // Python XY indexing is reverse from natural tensor indexing.  We premute here to match
```
Comment (cliffburdick) [inline include/matx/generators/meshgrid.h:85]: "permute"

File: include/matx/generators/meshgrid.h [hunk 3]
```diff
@@ -31,89 +31,89 @@
...
+        int perm[RANK];
+        for(int i=0; i < RANK; i++) {
+          perm[RANK-i-1] = i;
         }
-        static inline constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank() { return 2; }
-    };
-  }
+
+        // TODO transpose is not correct,  we need to reverse dimensions
```
Comment (cliffburdick) [inline include/matx/generators/meshgrid.h:91]: Is this comment old?

**Review by cliffburdick — APPROVED** [review]

File: include/matx/generators/meshgrid.h (hunk 1 above)
Comment (luitjens) [reply include/matx/generators/meshgrid.h:56]: a user will never hit this assert since they use a different api.  it was more for matx developers.

**Pull request merged**